src/claim/claim.js

```diff
@@ -90,38 +91,72 @@ export function num2buf2(num: number): Buffer {
 
 /**
  * Hash a string for a claim
-  * @param {string} elem
-  * @returns {Buffer} hash
+ * @param {string} elem
+ * @returns {Buffer} hash
  */
 export function hashString(s: string): Buffer {
   return utils.hashBytes(Buffer.from(s, 'utf8')).slice(1);
 }
 
 /**
  * Copy a buffer to an entry element ending at position start
-  * @param {Buffer} elem
-  * @param {number} start
-  * @param {Buffer} src
+ * @param {Buffer} elem
+ * @param {number} start
+ * @param {Buffer} src
  */
 export function copyToElemBuf(elem: Buffer, start: number, src: Buffer) {
   elem.fill(src, 32 - start - src.length, 32 - start);
 }
 
 /**
  * Get a buffer from an entry element ending at position start
-  * @param {Buffer} elem
-  * @param {number} start
-  * @param {number} length
+ * @param {Buffer} elem
+ * @param {number} start
+ * @param {number} length
  */
 export function getElemBuf(elem: Buffer, start: number, length: number): Buffer {
   return elem.slice(32 - start - length, 32 - start);
 }
 
+/**
+ * Set the most significant byte to 0
+ * @param {Buffer} elem - elem in big endian
+ * @returns {Buffer} hash with the first byte set to 0
+ */
+export function clearHighOrderByte(elem: Buffer): Buffer {
+  return elem.fill(0, 0, 1);
+}
+
+/**
+ * Check the 3 most significant bits are 0
+ * @param {Buffer} hash - hash in big endian
+ * @returns {void | Error} throws an error, void otherwise
+ */
+export function checkHashFitsClaim(hash: Buffer): void | Error {
```

Comment:
Having the other function `checkElemFitsClaim`, I would delete this one as is the same function but less accurate. If `checkHashFitsClaim` is called, could be the case that a value that is under BabyJub Q field don't passes this function comprovation, as don't has that values to zero.
Proposal: delete `checkHashFitsClaim` and use always `checkElemFitsClaim` (I'm taking about the content of the functions, not about the concrete names)

Reply:
Change done in commit 2b3c5ac405a686c1b10cf8f8847b4984c0362e07